secret-sharing/src/churp/dealer.rs

```diff
@@ -89,15 +119,41 @@ where
         SecretShare::new(x, p)
     }
 
-    /// Returns a random bivariate polynomial.
-    fn random_bivariate_polynomial(
+    /// Generates a random bivariate polynomial `B(x, y)` such that
+    /// the polynomials `B(x, y)`, `B(x, 0)`, and `B(0, y)` have non-zero
+    /// leading term, and the secret `B(0, 0)` is non-zero.
+    ///
+    /// This function is not constant time because it uses rejection
+    /// sampling to ensure that the polynomials have the maximum degree.
+    /// Additionally, the underlying prime field implementation may also
+    /// use rejection sampling to generate uniformly random elements.
+    fn generate_bivariate_polynomial(
         threshold: u8,
         rng: &mut impl RngCore,
     ) -> Result<BivariatePolynomial<G::Scalar>> {
         let deg_x = threshold;
         let deg_y = threshold.checked_mul(2).ok_or(Error::ThresholdTooLarge)?;
-        let bp = BivariatePolynomial::random(deg_x, deg_y, rng);
-        Ok(bp)
+
+        // When using a random RNG and a large prime field, this loop
+        // should execute once with an extremely high probability,
+        // so there is no need to optimize it by randomly selecting
+        // only the problematic coefficients.
+        loop {
```

Comment:
Should we still bound the loop to some number of iterations and fail? Just in the unlikely case the RNG is broken.

Reply:
Yes, we should.
- #5914